src/Workspaces/VisualBasic/Portable/Rename/LocalConflictVisitor.vb

```diff
@@ -155,7 +154,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Rename
                     ' is this local declared in the for or for each loop?
                     ' if not it was already added to the tracker before.
                     If local.IsFor OrElse local.IsForEach Then
-                        tokens.Add(identifierToken)
+                        tokens.Add(DirectCast(controlVariable, IdentifierNameSyntax).Identifier)
```

Comment:
how do we know the control variable is an identifier name?   Just because it's bound to a local?

Reply:
if it's not typed this way in the syntax model, i would not assume it.  For example, i believe VB can reference other values (including fields) in a for-loop.  So i'm pretty sure you could say something like `for me.i = 1 to 10` and have that be legal. 

(note: these would be good test cases).

Reply:
Indeed. Also, what about type characters? For `blah% = 1 To 10`? I forget how those are represented.

Reply:
OK, I have added a `SyntaxKind` check, although I'm still not convinced it could be anything other than `IdentifierNameSyntax` at this location. We reach the cast only when:
1. `controlVariable` is not an inline declaration
2. `controlVariable` is bind to a local
3. the local is declared in the loop

this should exclude the cases you mentioned above.

Reply:
@genlu You're *most likely* correct.  But i'd prefer us to be a bit paranoid here as that doesn't really cost us, and may save an unfortunate crash/investigation in the future.  Thanks! :D

Reply:
Check added.